### PR TITLE
Feature ini file settings take 2

### LINKED ISSUE #84: Custom key bindings
How did you imagine to implement custom key bindings? Will there be a menu, or maybe some sort of command line to change them? (I'd actually like to see a command line to access the config directly... And to be honest, having the config through #defines is rather crap, see #85)

### LINKED ISSUE #85: Proper config system
The current config system is obviously very alpha. Of cause the game needs a proper config that can be changed by the user, for example through a command line or even a GUI.
Are there any ideas yet? I'd have a few... Dunno how good they are, though.

### PR DESCRIPTION
**Pull request by nsg**
This is another version of PR #233 

* Use simpleini with structs to hold the data
* Clean up a little
* Make it possible to configure:
  * server credentials
  * server host / port
  * fov
  * fps
  * window size
  * radius
  * debug mode
  * key mappings
* Detect config path dependent on system/environ

fixes #85 and fixes #84 and closes #233

### CODE REVIEW

**Comment by nsg** [comment]
```ini
[client]
radius_start = 5
radius_max = 8
frames_per_second = 30
```

Is an improvement on my NUC, sure it lags a little but I can actually see things :)

I need to try to turn up `radius_max ` to 50 or something crazy on my gaming PC and see what happens :) (out of memory probably, but it will be interesting to see how long it survives)

**Comment by nsg** [comment]
What do you think @petterarvidsson?

File: lib/src/client.cpp [hunk 1]
```diff
@@ -41,23 +41,22 @@ namespace konstructs {
         return error_message;
     }
 
-    void Client::open_connection(const string &nick, const string &hash,
-                                 const string &hostname, const int port) {
+    void Client::open_connection(Settings settings) {
```
Comment (petterarvidsson) [inline lib/src/client.cpp]: Maybe it could take "Settings::Server" instead so we can remove "settings.server" in the code?

Comment (nsg) [inline lib/src/client.cpp]: Sounds good!

**Comment by nsg** [comment]
I have now tested it under Windows 10, works. The config file is created and loaded like under Linux. For the fun of it I tried to increase the max render distance but with a stable 60fps my gaming PC was still stuck at 608 (20).

**Pull request merged**